org.jacoco.examples/src/org/jacoco/examples/ExecutionDataServer.java

```diff
@@ -82,12 +82,17 @@ public void run() {
 			try {
 				while (reader.read()) {
 				}
-				socket.close();
 				synchronized (fileWriter) {
 					fileWriter.flush();
 				}
 			} catch (final IOException e) {
 				e.printStackTrace();
+			} finally {
```

Comment:
If you nest a try/finally in the try/catch we we need only one catch block.

Reply:
`reader.read()` may throw `IOException`, if it happens, socket.close() will not be executed, leaving the socket connection in `CLOSED_WAIT` status and thus result in resource leakage. 

The exception catch inside finally block is needed for `socket.close()`. 

Is there a way to use only one catch while keeping the above function?

Reply:
Here is my proposal:


		public void run() {
			try {
				try {
					while (reader.read()) {
					}
				} finally {
					socket.close();
				}
				synchronized (fileWriter) {
					fileWriter.flush();
				}
			} catch (final IOException e) {
				e.printStackTrace();
			}
		}

Reply:
@marchof Thank you for your review. 

I think there might be a small problem in the new version. 

If both reader.read() and `socket.close()` throws `IOException` (I see such case in other Closeable implementations, I don't know if it will happen in `socket.close()` or not), the `IOException` from `reader.read()` will be suppressed. 

I feel better to use a similar example in apache IOUtils. How do you think?

```
434     * Closeable closeable = null;
435     * try {
436     *     closeable = new FileReader(&quot;foo.txt&quot;);
437     *     // process closeable
438     *     closeable.close();
439     * } catch (Exception e) {
440     *     // error handling
441     * } finally {
442     *     IOUtils.closeQuietly(closeable);
443     * }
```

BTW, I think also better to move fileWriter.flush() out of the socket try block to make sure flush is executed always.

Reply:
This is didactic example code. So we want it to be as simple and readable as possible.

Reply:
I see. I think the current implementation example is enough. I will close the PR. Thank you.

Reply:
Should https://github.com/jacoco/jacoco/issues/1243 then also be closed?

Reply:
OK. I will close the issue.